fabric-rendering-v1/src/client/java/net/fabricmc/fabric/api/client/rendering/v1/HudRenderEvents.java

```diff
@@ -0,0 +1,138 @@
+			listener.afterMainHud(client, context, tickCounter);
+		}
+	});
+
+	/**
+	 * Called after the debug HUD, scoreboard, overlay message (action bar), and title and subtitle have been rendered, and before the {@link net.minecraft.client.gui.hud.ChatHud} is rendered.
+	 */
+	public static final Event<BeforeChat> BEFORE_CHAT = EventFactory.createArrayBacked(BeforeChat.class, listeners -> (client, context, tickCounter) -> {
+		for (BeforeChat listener : listeners) {
+			listener.beforeChat(client, context, tickCounter);
+		}
+	});
+
+	/**
+	 * Called after the entire HUD is rendered.
+	 */
+	public static final Event<Last> LAST = EventFactory.createArrayBacked(Last.class, listeners -> (client, context, tickCounter) -> {
+		for (Last listener : listeners) {
+			listener.onHudLast(client, context, tickCounter);
+		}
+	});
+
+	private HudRenderEvents() { }
+
+	@FunctionalInterface
+	public interface Start {
+		/**
+		 * Called at the start of HUD rendering, right before anything is rendered.
+		 *
+		 * @param client      the {@link MinecraftClient} instance
+		 * @param context     the {@link DrawContext} instance
+		 * @param tickCounter the {@link RenderTickCounter} instance with access to tick delta
+		 */
+		void onHudStart(MinecraftClient client, DrawContext context, RenderTickCounter tickCounter);
+	}
+
+	@FunctionalInterface
+	public interface AfterMiscOverlays {
+		/**
+		 * Called after misc overlays (vignette, spyglass, and powder snow) have been rendered, and before the crosshair is rendered.
+		 *
+		 * @param client      the {@link MinecraftClient} instance
+		 * @param context     the {@link DrawContext} instance
+		 * @param tickCounter the {@link RenderTickCounter} instance with access to tick delta
+		 */
+		void afterMiscOverlays(MinecraftClient client, DrawContext context, RenderTickCounter tickCounter);
+	}
+
+	@FunctionalInterface
+	public interface AfterMainHud {
+		/**
+		 * Called after the hotbar, status bars, and experience bar have been rendered, and before the status effects overlays are rendered.
+		 *
+		 * @param client      the {@link MinecraftClient} instance
+		 * @param context     the {@link DrawContext} instance
+		 * @param tickCounter the {@link RenderTickCounter} instance with access to tick delta
+		 */
+		void afterMainHud(MinecraftClient client, DrawContext context, RenderTickCounter tickCounter);
+	}
+
+	@FunctionalInterface
+	public interface BeforeChat {
+		/**
+		 * Called after the debug HUD, scoreboard, overlay message (action bar), and title and subtitle have been rendered, and before the {@link net.minecraft.client.gui.hud.ChatHud} is rendered.
+		 *
+		 * @param client      the {@link MinecraftClient} instance
+		 * @param context     the {@link DrawContext} instance
+		 * @param tickCounter the {@link RenderTickCounter} instance with access to tick delta
+		 */
+		void beforeChat(MinecraftClient client, DrawContext context, RenderTickCounter tickCounter);
+	}
```

Comment:
Maybe add `AFTER_SLEEP_OVERLAY` before this event? This would allow rendering before (so, behind) debug HUD, scoreboard, etc.

Reply:
You can just use `AFTER_MAIN_HUD`.

Reply:
Done in ce182ed.